### PR TITLE
DataGrid - A cell with a TextArea editor is closed when a mouse click is released outside the editor (T1161254)

### PR DESCRIPTION
**Pull request by Raushen**
TextArea editor closing cancelled (see https://supportcenter.devexpress.com/ticket/details/t1161254/datagrid-a-cell-with-a-textarea-editor-is-closed-when-a-mouse-click-is-released-outside)

### CODE REVIEW

**Review by pomahtri — DISMISSED** [review]

File: js/ui/grid_core/ui.grid_core.editing_cell_based.js [hunk 1]
```diff
@@ -65,6 +65,10 @@ export default {
                                 return $dropDownEditorOverlay.length > 0 && $componentElement.closest($dropDownEditorOverlay).length === 0;
                             }
 
+                            function isEditable($element) {
```
Comment (Alyar666) [inline js/ui/grid_core/ui.grid_core.editing_cell_based.js]: Is it possible to pull this method outside of the controller?

File: testing/tests/DevExpress.ui.widgets.dataGrid/editing.integration.tests.js [hunk 2]
```diff
@@ -4029,6 +4030,45 @@ QUnit.module('Editing', baseModuleConfig, () => {
...
+            },
+            onEditorPreparing(e) {
+                e.editorName = 'dxTextArea';
+            },
+        });
+        this.clock.tick(100);
+
+        $firstCell = $(dataGrid.getCellElement(0, 0));
+        $firstCell.trigger('dxclick');
+        this.clock.tick(5000);
```
Comment (Alyar666) [inline testing/tests/DevExpress.ui.widgets.dataGrid/editing.integration.tests.js]: Need such a big timeout?

File: testing/tests/DevExpress.ui.widgets.dataGrid/editing.integration.tests.js [hunk 3]
```diff
@@ -4029,6 +4030,45 @@ QUnit.module('Editing', baseModuleConfig, () => {
...
+
+        $firstCell = $(dataGrid.getCellElement(0, 0));
+        $textArea = $firstCell.find('textarea');
+        assert.equal($textArea.length, 1, 'textarea should appear');
+
+        $textArea.trigger($.Event('dxpointerdown', { pageX: $textArea.offset().left + 5, pageY: $textArea.offset().top + 5 }));
```
Comment (Alyar666) [inline testing/tests/DevExpress.ui.widgets.dataGrid/editing.integration.tests.js]: dxpointerdown and dxpointerup - are these events necessary?

File: testing/tests/DevExpress.ui.widgets.dataGrid/editing.integration.tests.js [hunk 4]
```diff
@@ -4029,6 +4030,45 @@ QUnit.module('Editing', baseModuleConfig, () => {
...
+        $textArea.trigger($.Event('dxpointerup', { pageX: 0, pageY: 0 }));
+        $textArea.trigger($.Event('click', { pageX: 0, pageY: 0, target: $('body').get(0) }));
```
Comment (Alyar666) [inline testing/tests/DevExpress.ui.widgets.dataGrid/editing.integration.tests.js]: pageX and pageY - are these parameters needed? I think they do not play an important role, and they can be left out.

File: testing/tests/DevExpress.ui.widgets.dataGrid/editing.integration.tests.js [hunk 5]
```diff
@@ -4029,6 +4030,45 @@ QUnit.module('Editing', baseModuleConfig, () => {
...
+        this.clock.tick(5000);
```
Comment (Alyar666) [inline testing/tests/DevExpress.ui.widgets.dataGrid/editing.integration.tests.js]: Need such a big timeout?

**Review by Alyar666 — CHANGES_REQUESTED** [review]

**Review by Alyar666 — APPROVED** [review]

**Review by williamvinogradov — APPROVED** [review]

**Pull request merged**